src/igvfd/tests/test_types_analysis_set.py

```diff
@@ -121,3 +121,15 @@ def test_protocols(testapp, analysis_set_base, measurement_set_with_protocols):
     )
     res = testapp.get(analysis_set_base['@id'])
     assert res.json.get('protocols') == ['https://www.protocols.io/test-protocols-url-12345']
+
+
+def test_functional_assay_mechanisms(testapp, analysis_set_base, measurement_set_with_functional_assay_mechanisms, phenotype_term_from_go):
+    testapp.patch_json(
+        analysis_set_base['@id'],
+        {
+            'input_file_sets': [measurement_set_with_functional_assay_mechanisms['@id']]
```

Comment:
Would it be a good idea to add two different measurement-sets with the same go_term? This way the with the test we can ensure we get a unique list of phenotype-terms on analysis-set.